### PR TITLE
BZ#1501483 - marked memory-based autoscaling as tech preview

### PR DESCRIPTION
**Pull request by gaurav-nelson**
For https://bugzilla.redhat.com/show_bug.cgi?id=1501483

### CODE REVIEW

File: dev_guide/pod_autoscaling.adoc [hunk 1]
```diff
@@ -215,6 +230,11 @@ spec:
 $ oc create -f hpa.yaml
 ----
 
+[NOTE]
+====
+Verify that your application's memory usage does not go down, which is more complicated than it seems.
```
Comment (gaurav-nelson) [inline dev_guide/pod_autoscaling.adoc]: @DirectXMan12 If we want to add a note for complexities we should include what is required? why it is required. 
```
Verify that your application's memory usage does not go down (go down from which value? the one we set in example?)
, which is more complicated than it seems. (why? what goes wrong if we don't do that)
```

Comment (gaurav-nelson) [reply dev_guide/pod_autoscaling.adoc]: @DirectXMan12 PTAL

Comment (DirectXMan12) [reply dev_guide/pod_autoscaling.adoc]: Probably something like

```
In order for memory-based autoscaling to work, memory usage must increase *and* decrease proportionally to replica count.  More specifically, an increase in replica count *must* lead to an overall decrease in memory (working set) usage per pod, on average.  Similarly, a decrease in replica count *must* lead to an overall increase in per-pod memory usage, on average.

It's important to ensure that your application meets these requirements before trying to use memory-base d autoscaling.  Use the OpenShift dashboard (and `oc top` in 3.10+) to verify the memory behavior of your application.

**Comment by gaurav-nelson** [comment]
@openshift/team-documentation PTAL 🆓

File: dev_guide/pod_autoscaling.adoc [hunk 2]
```diff
@@ -215,6 +230,21 @@ spec:
...
+====
+For memory-based autoscaling to work, memory usage must increase and decrease
+proportionally to the replica count. On average:
+
+* An increase in replica count must lead to an overall decrease in memory
+(working set) usage per-pod.
+* A decrease in replica count must lead to an overall increase in per-pod memory
+usage.
+
+Use the OpenShift dashboard to verify the memory behavior of your application
```
Comment (mburke5678) [inline dev_guide/pod_autoscaling.adoc]: What is the OpenShift dashboard? New in 3.9 and I am not aware? Should this be Kibana dashboard?

**Comment by mburke5678** [comment]
@gaurav-nelson A couple of nits.

File: dev_guide/pod_autoscaling.adoc [hunk 3]
```diff
@@ -215,6 +230,21 @@ spec:
...
+and ensure that your application meets these requirements before using
+memory-based autoscaling.
+====
```
Comment (mburke5678) [inline dev_guide/pod_autoscaling.adoc]: Do we need to tell users how to ensure the memory behavior?

File: dev_guide/pod_autoscaling.adoc (hunk 2 above)
Comment (DirectXMan12) [reply dev_guide/pod_autoscaling.adoc]: s/dashboard/webconsole/

File: dev_guide/pod_autoscaling.adoc (hunk 3 above)
Comment (gaurav-nelson) [reply dev_guide/pod_autoscaling.adoc]: Thanks @mburke5678, I am updating this to `.. use the web console to check the memory ...` I think this refers to manually increasing the replicas in web console and look at the memory graph.

**Comment by mburke5678** [comment]
@gaurav-nelson LGTM

**Pull request merged**